File: internal/bible.go
package baus

const (
	GENESIS             = 1
	EXODUS              = 1534
	LEVITICUS           = 2747
	NUMBERS             = 3606
	DEUTERONOMY         = 4894
	JOSHUA              = 5853
	JUDGES              = 6511
	RUTH                = 7129
	FIRSTSAMUEL         = 7214
	SECONDSAMUEL        = 8024
	FIRSTKINGS          = 8719
	SECONDKINGS         = 9535
	FIRSTCHRONICLES     = 10254
	SECONDCHRONICLES    = 11196
	EZRA                = 12018
	NEHEMIAH            = 12298
	ESTHER              = 12704
	JOB                 = 12871
	PSALMS              = 13941
	PROVERBS            = 16402
	ECCLESIASTES        = 17317
	SONGOFSOLOMON       = 17539
	ISAIAH              = 17656
	JEREMIAH            = 18948
	LAMENTATIONS        = 20312
	EZEKIEL             = 20466
	DANIEL              = 21739
	HOSEA               = 22096
	JOEL                = 22293
	AMOS                = 22366
	OBADIAH             = 22512
	JONAH               = 22533
	MICAH               = 22581
	NAHUM               = 22686
	HABAKKUK            = 22733
	ZEPHANIAH           = 22789
	HAGGAI              = 22842
	ZECHARIAH           = 22880
	MALACHI             = 23091
	MATTHEW             = 23146
	MARK                = 24217
	LUKE                = 24895
	JOHN                = 26046
	ACTS                = 26925
	ROMANS              = 27932
	FIRSTCORINTHIANS    = 28365
	SECONDCORINTHIANS   = 28802
	GALATIANS           = 29059
	EPHESIANS           = 29208
	PHILIPPIANS         = 29363
	COLOSSIANS          = 29467
	FIRSTTHESSALONIANS  = 29562
	SECONDTHESSALONIANS = 29651
	FIRSTTIMOTHY        = 29698
	SECONDTIMOTHY       = 29811
	TITUS               = 29894
	PHILEMON            = 29940
	HEBREWS             = 29965
	JAMES               = 30268
	FIRSTPETER          = 30376
	SECONDPETER         = 30481
	FIRSTJOHN           = 30542
	SECONDJOHN          = 30647
	THIRDJOHN           = 30660
	JUDE                = 30674
	REVELATION          = 30699
)

func getOrdinalBookNumber(book string) int {
	switch book {
	case "GENESIS":
		return GENESIS
	case "EXODUS":
		return EXODUS
	case "LEVITICUS":
		return LEVITICUS
	case "NUMBERS":
		return NUMBERS
	case "DEUTERONOMY":
		return DEUTERONOMY
	case "JOSHUA":
		return JOSHUA
	case "JUDGES":
		return JUDGES
	case "RUTH":
		return RUTH
	case "FIRSTSAMUEL":
		return FIRSTSAMUEL
	case "SECONDSAMUEL":
		return SECONDSAMUEL
	case "FIRSTKINGS":
		return FIRSTKINGS
	case "SECONDKINGS":
		return SECONDKINGS
	case "FIRSTCHRONICLES":
		return FIRSTCHRONICLES
	case "SECONDCHRONICLES":
		return SECONDCHRONICLES
	case "EZRA":
		return EZRA
	case "NEHEMIAH":
		return NEHEMIAH
	case "ESTHER":
		return ESTHER
	case "JOB":
		return JOB
	case "PSALMS":
		return PSALMS
	case "PROVERBS":
		return PROVERBS
	case "ECCLESIASTES":
		return ECCLESIASTES
	case "SONGOFSOLOMON":
		return SONGOFSOLOMON
	case "ISAIAH":
		return ISAIAH
	case "JEREMIAH":
		return JEREMIAH
	case "LAMENTATIONS":
		return LAMENTATIONS
	case "EZEKIEL":
		return EZEKIEL
	case "DANIEL":
		return DANIEL
	case "HOSEA":
		return HOSEA
	case "JOEL":
		return JOEL
	case "AMOS":
		return AMOS
	case "OBADIAH":
		return OBADIAH
	case "JONAH":
		return JONAH
	case "MICAH":
		return MICAH
	case "NAHUM":
		return NAHUM
	case "HABAKKUK":
		return HABAKKUK
	case "ZEPHANIAH":
		return ZEPHANIAH
	case "HAGGAI":
		return HAGGAI
	case "ZECHARIAH":
		return ZECHARIAH
	case "MALACHI":
		return MALACHI
	case "MATTHEW":
		return MATTHEW
	case "MARK":
		return MARK
	case "LUKE":
		return LUKE
	case "JOHN":
		return JOHN
	case "ACTS":
		return ACTS
	case "ROMANS":
		return ROMANS
	case "FIRSTCORINTHIANS":
		return FIRSTCORINTHIANS
	case "SECONDCORINTHIANS":
		return SECONDCORINTHIANS
	case "GALATIANS":
		return GALATIANS
	case "EPHESIANS":
		return EPHESIANS
	case "PHILIPPIANS":
		return PHILIPPIANS
	case "COLOSSIANS":
		return COLOSSIANS
	case "FIRSTTHESSALONIANS":
		return FIRSTTHESSALONIANS
	case "SECONDTHESSALONIANS":
		return SECONDTHESSALONIANS
	case "FIRSTTIMOTHY":
		return FIRSTTIMOTHY
	case "SECONDTIMOTHY":
		return SECONDTIMOTHY
	case "TITUS":
		return TITUS
	case "PHILEMON":
		return PHILEMON
	case "HEBREWS":
		return HEBREWS
	case "JAMES":
		return JAMES
	case "FIRSTPETER":
		return FIRSTPETER
	case "SECONDPETER":
		return SECONDPETER
	case "FIRSTJOHN":
		return FIRSTJOHN
	case "SECONDJOHN":
		return SECONDJOHN
	case "THIRDJOHN":
		return THIRDJOHN
	case "JUDE":
		return JUDE
	case "REVELATION":
		return REVELATION
	default:
		return 0
	}
}

File: html/js/bible.js
const books = [{'book_name': 'GENESIS', 'ordinal_verse_start': 1508, 'chapters': 50}, {'book_name': 'EXODUS', 'ordinal_verse_start': 2709, 'chapters': 40}, {'book_name': 'LEVITICUS', 'ordinal_verse_start': 3572, 'chapters': 27}, {'book_name': 'NUMBERS', 'ordinal_verse_start': 4881, 'chapters': 36}, {'book_name': 'DEUTERONOMY', 'ordinal_verse_start': 5841, 'chapters': 34}, {'book_name': 'JOSHUA', 'ordinal_verse_start': 6478, 'chapters': 24}, {'book_name': 'JUDGES', 'ordinal_verse_start': 7104, 'chapters': 21}, {'book_name': 'RUTH', 'ordinal_verse_start': 7192, 'chapters': 4}, {'book_name': '1SAMUEL', 'ordinal_verse_start': 8011, 'chapters': 31}, {'book_name': '2SAMUEL', 'ordinal_verse_start': 8694, 'chapters': 24}, {'book_name': '1KINGS', 'ordinal_verse_start': 9482, 'chapters': 22}, {'book_name': '2KINGS', 'ordinal_verse_start': 10224, 'chapters': 25}, {'book_name': '1CHRONICLES', 'ordinal_verse_start': 11166, 'chapters': 29}, {'book_name': '2CHRONICLES', 'ordinal_verse_start': 11995, 'chapters': 36}, {'book_name': 'EZRA', 'ordinal_verse_start': 12254, 'chapters': 10}, {'book_name': 'NEHEMIAH', 'ordinal_verse_start': 12673, 'chapters': 13}, {'book_name': 'ESTHER', 'ordinal_verse_start': 12868, 'chapters': 10}, {'book_name': 'JOB', 'ordinal_verse_start': 13924, 'chapters': 42}, {'book_name': 'PSALMS', 'ordinal_verse_start': 16396, 'chapters': 150}, {'book_name': 'PROVERBS', 'ordinal_verse_start': 17286, 'chapters': 31}, {'book_name': 'ECCLESIASTES', 'ordinal_verse_start': 17525, 'chapters': 12}, {'book_name': 'SONG OF SOLOMON', 'ordinal_verse_start': 17642, 'chapters': 8}, {'book_name': 'ISAIAH', 'ordinal_verse_start': 18924, 'chapters': 66}, {'book_name': 'JEREMIAH', 'ordinal_verse_start': 20278, 'chapters': 52}, {'book_name': 'LAMENTATIONS', 'ordinal_verse_start': 20444, 'chapters': 5}, {'book_name': 'EZEKIEL', 'ordinal_verse_start': 21704, 'chapters': 48}, {'book_name': 'DANIEL', 'ordinal_verse_start': 22083, 'chapters': 12}, {'book_name': 'HOSEA', 'ordinal_verse_start': 22284, 'chapters': 14}, {'book_name': 'JOEL', 'ordinal_verse_start': 22345, 'chapters': 3}, {'book_name': 'AMOS', 'ordinal_verse_start': 22497, 'chapters': 9}, {'book_name': 'OBADIAH', 'ordinal_verse_start': 22512, 'chapters': 1}, {'book_name': 'JONAH', 'ordinal_verse_start': 22570, 'chapters': 4}, {'book_name': 'MICAH', 'ordinal_verse_start': 22666, 'chapters': 7}, {'book_name': 'NAHUM', 'ordinal_verse_start': 22714, 'chapters': 3}, {'book_name': 'HABAKKUK', 'ordinal_verse_start': 22770, 'chapters': 3}, {'book_name': 'ZEPHANIAH', 'ordinal_verse_start': 22822, 'chapters': 3}, {'book_name': 'HAGGAI', 'ordinal_verse_start': 22857, 'chapters': 2}, {'book_name': 'ZECHARIAH', 'ordinal_verse_start': 23070, 'chapters': 14}, {'book_name': 'MALACHI', 'ordinal_verse_start': 23140, 'chapters': 4}, {'book_name': 'MATTHEW', 'ordinal_verse_start': 24197, 'chapters': 28}, {'book_name': 'MARK', 'ordinal_verse_start': 24875, 'chapters': 16}, {'book_name': 'LUKE', 'ordinal_verse_start': 25993, 'chapters': 24}, {'book_name': 'JOHN', 'ordinal_verse_start': 26900, 'chapters': 21}, {'book_name': 'ACTS', 'ordinal_verse_start': 27901, 'chapters': 28}, {'book_name': 'ROMANS', 'ordinal_verse_start': 28338, 'chapters': 16}, {'book_name': '1CORINTHIANS', 'ordinal_verse_start': 28778, 'chapters': 16}, {'book_name': '2CORINTHIANS', 'ordinal_verse_start': 29045, 'chapters': 13}, {'book_name': 'GALATIANS', 'ordinal_verse_start': 29190, 'chapters': 6}, {'book_name': 'EPHESIANS', 'ordinal_verse_start': 29339, 'chapters': 6}, {'book_name': 'PHILIPPIANS', 'ordinal_verse_start': 29444, 'chapters': 4}, {'book_name': 'COLOSSIANS', 'ordinal_verse_start': 29544, 'chapters': 4}, {'book_name': '1THESSALONIANS', 'ordinal_verse_start': 29623, 'chapters': 5}, {'book_name': '2THESSALONIANS', 'ordinal_verse_start': 29680, 'chapters': 3}, {'book_name': '1TIMOTHY', 'ordinal_verse_start': 29790, 'chapters': 6}, {'book_name': '2TIMOTHY', 'ordinal_verse_start': 29872, 'chapters': 4}, {'book_name': 'TITUS', 'ordinal_verse_start': 29925, 'chapters': 3}, {'book_name': 'PHILEMON', 'ordinal_verse_start': 29940, 'chapters': 1}, {'book_name': 'HEBREWS', 'ordinal_verse_start': 30243, 'chapters': 13}, {'book_name': 'JAMES', 'ordinal_verse_start': 30356, 'chapters': 5}, {'book_name': '1PETER', 'ordinal_verse_start': 30467, 'chapters': 5}, {'book_name': '2PETER', 'ordinal_verse_start': 30524, 'chapters': 3}, {'book_name': '1JOHN', 'ordinal_verse_start': 30626, 'chapters': 5}, {'book_name': '2JOHN', 'ordinal_verse_start': 30647, 'chapters': 1}, {'book_name': '3JOHN', 'ordinal_verse_start': 30660, 'chapters': 1}, {'book_name': 'JUDE', 'ordinal_verse_start': 30674, 'chapters': 1}, {'book_name': 'REVELATION', 'ordinal_verse_start': 31082, 'chapters': 22}]

const bookButton = document.getElementById('book-choose');
const chapterButton = document.getElementById('chapter-choose');
for(i=0; i<books.length; i++){
    a = document.createElement('option');
    a.textContent = books[i].book_name;
    bookButton.appendChild(a);
}
// its gonna start with Genesis so make chapter size 50
for(i=1; i<=50;i++){
    a = document.createElement('option');
    a.textContent = i
    chapterButton.appendChild(a);
}
// books.forEach(element => console.log(element));
const bookChoose = document.getElementById('book-choose');
const poemDisplay = document.getElementById('scripture');
const bookName = bookButton.value;
const chapterNum = chapterButton.value;
const searchHandler = document.getElementById('search-text');

console.log("bookname: ", bookName);
console.log("chaper: ", chapterNum);

bookChoose.addEventListener('change', () => {
    // clear all elements in scripture area
    document.getElementById('scripture').innerHTML = '';
    // set the verse value
    const placeholder = `https://mintz5.duckdns.org/bible/${bookButton.value}/${chapterButton.value}`;
    console.log(`url: ${placeholder}`);
    // updateDisplay(placeholder);
    const clearText = poemDisplay.querySelectorAll('p');
    clearText.forEach(function(item){
        item.remove();
    })
    // populate the text with Chapter 1 since its new book
    fetch(`https://mintz5.duckdns.org/bible/${bookButton.value}/1`)
        .then(response => {
            if(!response.ok){
                throw new Error(`HTTP Error: ${response.status}`);
            }
            return response.json();
        })
        .then(function (data){
            for(i=0; i<data.length;i++){
                const p = document.createElement('p');
                p.textContent = `${i+1}: ${data[i].text}`;
                poemDisplay.appendChild(p);
            }
        })
    // update the chapter selection
    for(i=0; i<books.length; i++){
        if(books[i].book_name == bookButton.value){
            let chapterMax = books[i].chapters
            chapterHandle = chapterButton.querySelectorAll('option');
            chapterHandle.forEach(function(item){
                item.remove();
            })
            for(j=1; j<=chapterMax; j++){
                co = document.createElement('option');
                co.textContent = j;
                chapterButton.appendChild(co);
            }
        }
    }
    // update the title to match book name
    document.getElementById('theTitle').textContent = bookButton.value;
});

chapterButton.addEventListener('change', () =>{
    // clear scripture area
    document.getElementById('scripture').innerHTML = '';
    const titleHandle = document.getElementById('theTitle');
    titleHandle.textContent = bookButton.value;
    const verse = bookChoose.value;
    const placeholder = `https://mintz5.duckdns.org/bible/${bookButton.value}/${chapterButton.value}`;
    console.log(verse);
    console.log(`url: ${placeholder}`);
    // updateDisplay(placeholder);
    const clearPoemDisplay = poemDisplay.querySelectorAll('p');
    clearPoemDisplay.forEach(function (item){
        item.remove();
    })
    fetch(placeholder)
        .then(response => {
            console.log("calling url");
            if(!response.ok){
                throw new Error(`HTTP error: ${response.status}`);
            }
            return response.json();
        })
        .then(function (data){
            for(i=0; i<data.length;i++){
                const p = document.createElement('p');
                p.textContent = `${i+1}: ${data[i].text}`;
                poemDisplay.appendChild(p);
            }
        })
})

function updateDisplay(verse){
    // clear the display area
    const titleHandle = document.getElementById('theTitle');
    titleHandle.textContent = bookButton.value;
    const clearPoemDisplay = poemDisplay.querySelectorAll('p');
    clearPoemDisplay.forEach(function (item){
        item.remove();
    })
    fetch(verse)
        .then(response => {
            console.log("calling url");
            if(!response.ok){
                throw new Error(`HTTP error: ${response.status}`);
            }
            return response.json();
        })
        .then(function (data){
            for(i=0; i<data.length;i++){
                const p = document.createElement('p');
                p.textContent = `${i+1}: ${data[i].text}`;
                poemDisplay.appendChild(p);
            }
        })
    // update chapter selection count
    clearChapters = chapterButton.querySelectorAll('option');
    clearChapters.forEach(function(item){
        item.remove();
    })
    // update chapters with respective book
    console.log("POPULATING CHAPTERS NOW");
    for(i=0; i<books.length;i++){
        console.log("looking at book ", books[i].book_name);
        if (books[i].book_name == bookButton.value){
            console.log(`found book: ${books[i].book_name}`)
            for(j=1; j<=books[i].chapters; j++){
                o = document.createElement('option');
                o.textContent= j;
                chapterButton.appendChild(o);
            }
            break;
        }
    }
} 
